### PR TITLE
BOOKKEEPER-841: Bookie should calculate ledgers map writing a new entry log file

### PR DESCRIPTION
**Pull request by merlimat**
@sijie I've addressed all comments from https://reviews.apache.org/r/33061

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java [hunk 1]
```diff
@@ -101,14 +114,60 @@ public long getLogId() {
     private final CopyOnWriteArrayList<EntryLogListener> listeners
         = new CopyOnWriteArrayList<EntryLogListener>();
 
+    private static final int HEADER_V0 = 0; // Old log file format (no ledgers map index)
+    private static final int HEADER_V1 = 1; // Unrecognized header version
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java]: the comment is wrong?

**Comment by sijie** [comment]
The overall of this change looks good to me. only a minor comment.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java (hunk 1 above)
Comment (merlimat) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java]: True, I left the wrong comment when refactoring.

**Comment by sijie** [comment]
the change looks good to me +1

**Pull request closed without merging**